app/lib/onboarding/pages/page.dart

```diff
@@ -0,0 +1,267 @@
+
+  @override
+  Widget build(BuildContext context) {
+    final colors = _pages.map((page) => page.color);
+    final tweenSequenceItems = <TweenSequenceItem>[];
+    for (var tweenIndex = 0; tweenIndex < colors.length - 1; tweenIndex++) {
+      tweenSequenceItems.add(
+        TweenSequenceItem(
+          weight: 1,
+          tween: ColorTween(
+            begin: colors.elementAt(tweenIndex),
+            end: colors.elementAt(tweenIndex + 1),
+          ),
+        ),
+      );
+    }
+    final background = TweenSequence(tweenSequenceItems);
+
+    final pageController = usePageController(initialPage: 0);
+    final currentPage = useState(0);
+
+    return Scaffold(
+      body: AnimatedBuilder(
+        animation: pageController,
+        builder: (context, child) {
+          final color = pageController.hasClients
+              ? pageController.page! / (_pages.length - 1)
+              : .0;
+
+          return DecoratedBox(
+            decoration: BoxDecoration(
+              color: background.evaluate(AlwaysStoppedAnimation(color)),
+            ),
+            child: child,
+          );
+        },
+        child: Stack(alignment: Alignment.topCenter, children: [
+          Positioned.fill(
+            child: PageView(
+              controller: pageController,
+              onPageChanged: (newPage) => currentPage.value = newPage,
+              children: _pages,
+            ),
+          ),
+          Positioned(
+            bottom: 80,
+            child: Row(
+              mainAxisAlignment: MainAxisAlignment.center,
+              children: _buildPageIndicator(context, currentPage.value),
+            ),
+          ),
+          Positioned(
+            bottom: 16,
+            right: 16,
+            child: _buildNextButton(
+              context,
+              pageController,
+              currentPage.value == _pages.length - 1,
+            ),
+          ),
+        ]),
+      ),
+    );
+  }
+
+  List<Widget> _buildPageIndicator(BuildContext context, int currentPage) {
+    final list = <Widget>[];
+    for (var i = 0; i < _pages.length; ++i) {
+      list.add(_indicator(context, i == currentPage));
+    }
+    return list;
+  }
+
+  Widget _indicator(BuildContext context, bool isActive) {
+    return AnimatedContainer(
+      duration: Duration(milliseconds: 150),
+      margin: EdgeInsets.symmetric(horizontal: 8),
+      height: 8,
+      width: isActive ? 24 : 16,
+      decoration: BoxDecoration(
+        color: isActive ? Colors.white : PharmeTheme.onSurfaceColor,
+        borderRadius: BorderRadius.all(Radius.circular(12)),
+      ),
+    );
+  }
+
+  Widget _buildNextButton(
+    BuildContext context,
+    PageController pageController,
+    bool isLastPage,
+  ) {
+    return TextButton(
+      onPressed: () {
+        if (isLastPage) {
+          if (_isLoggedIn) {
+            context.router.pop();
+          } else {
+            context.router.replace(LoginRouter());
+          }
```

Comment:
```suggestion
          _isLoggedIn ? context.router.pop() : context.router.replace(LoginRouter());
  
```
I would use tenary operator for the inner statement, it's smh weird with this if-else just for one statement to be executed in the respective case